### ISSUE TITLE
BIG-IP clustering fails when a gtm server exists on the BIG-IP

### ISSUE THREAD

**Issue by cbascom** [F5Networks/f5-declarative-onboarding#201]
[restnoded.log](https://github.com/F5Networks/f5-declarative-onboarding/files/5882036/restnoded.log)
### Environment
 * Declarative Onboarding Version: 1.18.0-4
 * BIG-IP Version: 16.0.0.1-0.0.3

### Summary
When attempting to cluster BIG-IPs that have a gtm server on them, the processing of the DO declaration is failing. We are not actually attempting to do anything with GTM (DNS), but at least in some cases it appears that a default gtm server has been created on the BIG-IP. DO sees this gtm server and attempts to modify it but that modification fails I believe because one or more mandatory parameters are being omitted.

This is a blocking issue for our environment. I have had to downgrade to 1.17.0-3 due to this issue.

### Steps To Reproduce
1. Ensure gtm is provisioned and that a default gtm server exists on one or both of the BIG-IPs
2. Submit the following declaration:
```json
{
   "async":true,
   "schemaVersion":"1.18.0",
   "Common":{
      "trust":{
         "localUsername":"admin",
         "remoteHost":"10.10.36.237",
         "localPassword":"admin",
         "remotePassword":"admin",
         "remoteUsername":"admin",
         "class":"DeviceTrust"
      },
      "f5test-ha-dg":{
         "autoSync":true,
         "fullLoadOnSync":false,
         "saveOnAutoSync":false,
         "members":[
            "10.10.36.237",
            "10.10.36.215"
         ],
         "asmSync":false,
         "owner":"/Common/trust/remoteHost",
         "networkFailover":true,
         "type":"sync-failover",
         "class":"DeviceGroup"
      },
      "failoverAddress":{
         "class":"FailoverUnicast",
         "address":"10.10.36.237"
      },
      "class":"Tenant",
      "configsync":{
         "configsyncIp":"10.10.36.237",
         "class":"ConfigSync"
      }
   },
   "class":"Device",
   "label":"BVT BIG-IP declaration for DO"
}
```

2. Observe the following error response:
```json
{
   "id":"8092a4e0-cec7-4934-92e0-a56125233757",
   "selfLink":"https://localhost/mgmt/shared/declarative-onboarding/task/8092a4e0-cec7-4934-92e0-a56125233757",
   "code":500,
   "status":"ERROR",
   "message":"rollback failed",
   "errors":[
      "one or more configuration identifiers must be provided",
      "unicastAddresses.map is not a function"
   ],
   "result":{
      "class":"Result",
      "code":500,
      "status":"ERROR",
      "message":"rollback failed",
      "errors":[
         "one or more configuration identifiers must be provided",
         "unicastAddresses.map is not a function"
      ]
   },
   "declaration":{
      "async":true,
      "schemaVersion":"1.18.0",
      "Common":{
         "trust":{
            "localUsername":"admin",
            "remoteHost":"10.10.36.237",
            "remoteUsername":"admin",
            "class":"DeviceTrust"
         },
         "f5test-ha-dg":{
            "autoSync":true,
            "fullLoadOnSync":false,
            "saveOnAutoSync":false,
            "members":[
               "10.10.36.237",
               "10.10.36.215"
            ],
            "asmSync":false,
            "owner":"/Common/trust/remoteHost",
            "networkFailover":true,
            "type":"sync-failover",
            "class":"DeviceGroup"
         },
         "failoverAddress":{
            "class":"FailoverUnicast",
            "address":"10.10.36.215",
            "port":1026
         },
         "class":"Tenant",
         "configsync":{
            "configsyncIp":"10.10.36.215",
            "class":"ConfigSync"
         }
      },
      "class":"Device",
      "label":"BVT BIG-IP declaration for DO"
   }
}
```

### Expected Behavior
The BIG-IPs should have been clustered successfully. Nothing should have been done with gtm (DNS) since we were not trying to configure anything there.

### Actual Behavior
The DO declaration failed to be processed successfully. It first failed with a 'one or more configuration identifiers must be provided' error when attempting to modify the GSLB server. It then failed the rollback with a 'unicastAddresses.map is not a function' error.

This is the gtm server that caused the issue:
```
[root@h024-bp1-36-215:TimeLimitedModules::Standby:Disconnected] config # tmsh list gtm server
gtm server h024-bp1-36-215.mgmt.pdsea.f5net.com {
    addresses {
        10.13.36.215 {
            device-name /Common/bigip1
        }
    }
    datacenter Default_DC
    devices {
        bigip1 {
            addresses {
                10.13.36.215 { }
            }
        }
    }
    monitor bigip
    product bigip
    virtual-server-discovery enabled
}
```

I have attached the [restnoded.log](https://github.com/F5Networks/f5-declarative-onboarding/files/5882044/restnoded.log) where this failed. This is the error line from that file:
```
Wed, 27 Jan 2021 16:53:30 GMT - severe: [f5-declarative-onboarding: gslbHandler.js] Error creating Servers: one or more configuration identifiers must be provided
```

**Comment by dstokesf5** [comment]
@cbascom: Have you tried adding Default_DC to your DO declaration? Also, do you actually want/need this GTM server? I know you were unable to narrow down where it was coming from, but I suspect DO should be trying to delete that server instead of trying to create/modify it.

**Comment by cbascom** [comment]
@dstokesf5 I have not tried adding it to the declaration, I can give that a shot hopefully next week.

We do not want/need this GTM server, so DO removing it would work from our standpoint. I just don't like making a call on doing something like that without understanding what is creating it in the first place. How does DO handle pre-existing configuration in other cases? Like if some self IPs already exist on the system and none are specified in the declaration, does it delete all of them? If so, seems like removing this GTM server would make sense too.

**Comment by cbascom** [comment]
I just ran into this in our scale environment as well. Below is the default server that got created in case that is helpful. I'm going to try and put the server into the declaration to see if that is a valid workaround, but not sure why it chose the IPv6 addresses instead of the IPv4 addresses when creating this default server.

```
gtm server bip-10-255-14-22.f5net.com {
    addresses {
        fd32:f5:0:a70::e16 {
            device-name /Common/bigip1
        }
    }
    datacenter Default_DC
    devices {
        bigip1 {
            addresses {
                fd32:f5:0:a70::e16 { }
            }
        }
    }
    monitor bigip
    product bigip
    virtual-server-discovery enabled
}
```

**Comment by cbascom** [comment]
@dstokesf5 It is not currently possible to specify that GTM server via DO, so that is out as a workaround. The problem is that periods are not allowed in the gtm server name, so the [attached declaration](https://github.com/F5Networks/f5-declarative-onboarding/files/6052839/data.txt) fails with this error:

> Fri, 26 Feb 2021 21:35:28 GMT - info: [f5-declarative-onboarding: restWorker.js] Bad declaration: [{"keyword":"pattern","dataPath":".declaration.Common","schemaPath":"#/properties/Common/propertyNames/pattern","params":{"pattern":"^[A-Za-z][0-9A-Za-z_-]{0,47}$"},"message":"should match pattern \"^[A-Za-z][0-9A-Za-z_-]{0,47}$\"","propertyName":"bip-10-255-42-41.f5net.com"},{"keyword":"propertyNames","dataPath":".declaration.Common","schemaPath":"#/properties/Common/propertyNames","params":{"propertyName":"bip-10-255-42-41.f5net.com"},"message":"property name 'bip-10-255-42-41.f5net.com' is invalid"}]

**Comment by dstokesf5** [comment]
I have added AUTOTOOL-2224 to our internal product backlog. In this ticket we want to make sure that the server can be declared if the user wishes to keep it, and that it will be properly deleted if the user excludes it from their declaration.

**Comment by dstokesf5** [comment]
This issue was resolved as part of the 1.21.0 release.
